client/src/containers/approver/proposals/TableList.js

```diff
@@ -0,0 +1,340 @@
+/* Copyright 2019 Contributors to Hyperledger Sawtooth
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+----------------------------------------------------------------------------- */
+
+
+import React, { Component } from 'react';
+import {
+  Checkbox,
+  Header,
+  Icon,
+  Table } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+
+
+import './TableList.css';
+import * as utils from 'services/Utils';
+import Avatar from 'components/layouts/Avatar';
+
+
+/**
+ *
+ * @class         TableList
+ * @description   Displays roles in a list when approving proposals
+ *
+ */
+class TableList extends Component {
+
+  static propTypes = {
+    getRoles:              PropTypes.func,
+    getUsers:              PropTypes.func,
+    handleChange:          PropTypes.func,
+    openProposalsByRole:   PropTypes.object,
+    openProposalsByUser:   PropTypes.object,
+    roleFromId:            PropTypes.func,
+    roles:                 PropTypes.array,
+    selectedProposals:     PropTypes.array,
+    selectedRoles:         PropTypes.array,
+    users:                 PropTypes.array,
+  };
+
+
+  state = { column: null, direction: null };
+
+
+  // TODO: Refactor
```

Comment:
Is there a ticket for this TODO or will it be handled soon?

Reply:
Created #1307.